src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -472,6 +472,7 @@
 
         :else
         {:status 200
+         :headers {"Content-Type" "application/octet-stream"}
```

Comment:
The success case should always be `application/octet-stream`; for the above error cases it seems reasonable to return `text/plain`. I was musing about the slightly wider case - if a given response doesn't have a content-type at all should we add one?